ScraperFC/SofaScore.py

```diff
@@ -1,144 +1,454 @@
+
+        df = pd.DataFrame()
+        for i in range(len(response.json()['statistics'][0]['groups'])):
+            df_valores = pd.DataFrame(response.json()['statistics'][0]['groups'][i]['statisticsItems'])
+            df = pd.concat([df,df_valores])
+        df = df[['name', 'home', 'homeValue', 'homeTotal','away', 'awayValue', 'awayTotal']]
+        return df
+    
+    def get_players_match_stats(self, match_url):
+        """Returns match data for each player.
+
+        Args:
+            match_url (str): Full link to a SofaScore match
+
+        Returns:
+            DataFrames: A DataFrame for home and another one for away teams with each row being a player and in each columns a different statistic or data of the player
+        """
+
+        headers = self.get_requests_headers()
+        match_id = self.get_match_id(match_url)
+        
+        response = requests.get(f'https://api.sofascore.com/api/v1/event/{match_id}/lineups', headers=headers)
+
+        home_team, away_team = self.get_team_names(match_url)
+        names = {'home': home_team, 'away': away_team}
+
+        teams = ['home', 'away']
+        dataframes = {}
+        for team in teams:
+            dataframes[f'df_{team}'] = pd.DataFrame()
+            data = pd.DataFrame(response.json()[team]['players'])
+            columns_list = [data['player'].apply(pd.Series), data['shirtNumber'], data['jerseyNumber'], data['position'], data['substitute'], data['statistics'].apply(pd.Series), data['captain']]
+            dataframes[f'df_{team}'] = pd.concat(columns_list, axis=1)
+            dataframes[f'df_{team}']['team'] = names[team]
+        
+        return dataframes['df_home'], dataframes['df_away']
+    
+    def get_players_average_positions(self, match_url):
+        """Return player averages positions for each team
+
+        Args:
+            match_url (str): Full link to a SofaScore match
+
+        Returns:
+            DataFrame: Each row is a player and columns averageX and averageY denote their average position on the match.
+        """
+        match_id = self.get_match_id(match_url)
+        headers = self.get_requests_headers()
+
+        response = requests.get(f'https://api.sofascore.com/api/v1/event/{match_id}/average-positions', headers=headers)
+        response.json()
+        home_team, away_team = self.get_team_names(match_url)
+        names = {'home': home_team, 'away': away_team}
+        teams = ['home', 'away']
+        dataframes = {}
+        for team in teams:
+            df_team_averages = pd.DataFrame(response.json()[team])
+            df_team_averages['team'] = names[team]
+            dataframes[f'df_{team}'] = pd.concat([df_team_averages['player'].apply(pd.Series), df_team_averages.drop(columns=['player'])],axis=1)
+            
+        return dataframes['df_home'], dataframes['df_away']
+    
+    def get_player_heatmap(self, match_url, player, save_heatmap=True, cmap='OrRd'):
```

Comment:
Question, not a comment: did you see an API call to get the x-y coordinates for team heatmaps? Or would we need to scrape and concatenate coordinates for all players for each team?